### PR TITLE
conditional update location by name, addressed git hub comments

### PR DESCRIPTION
**Pull request by sashrika**
https://issues.openmrs.org/browse/FM-101

please close the pull request https://github.com/openmrs/openmrs-module-fhir/pull/89

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java [hunk 1]
```diff
@@ -109,33 +110,68 @@ public void deleteLocation(@IdParam IdDt theId) {
...
 		if (theConditional != null) {
-			int startIndex = theConditional.lastIndexOf('=');
-			id = theConditional.substring(startIndex + 1);
+			List<Location> locationList = null;
+			String locationName = null;
+			try {
+				int startIndex = theConditional.lastIndexOf('=');
+				locationName = theConditional.substring(startIndex + 1);
+			}
+			catch (Exception e) { // will catch nullpointerexceptions and indexoutofboundexceptions
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: catch  need to be align with 
```java
} catch(Exception e)
```java

But here is wise thing to do is check the null pointer. What's the array out of bounds here? 
We are not catching generic exceptions

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java [hunk 2]
```diff
@@ -109,33 +110,68 @@ public void deleteLocation(@IdParam IdDt theId) {
...
-			int startIndex = theConditional.lastIndexOf('=');
-			id = theConditional.substring(startIndex + 1);
+			List<Location> locationList = null;
+			String locationName = null;
+			try {
+				int startIndex = theConditional.lastIndexOf('=');
+				locationName = theConditional.substring(startIndex + 1);
+			}
+			catch (NullPointerException e) {
+				operationoutcome = new OperationOutcome();
```
Comment (sashrika) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: this is what we get from the openmrs formatter @harsha89

Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: oopz i think something is wrong there. Can you fix it :). If you use intelllij change this rule

Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: No no don't catch nul pointer. Just have a if condition :)

Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: avoid catching generic exceptions

Comment (sashrika) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: I am using eclipse, any way I'll look into this format issue. 
Yeah I could check the null. This happens when you carelessly copy and past code :)

Thanks!

Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: ha ha there are some other places :dancer: you going to fix them :wink:  :laughing:

Comment (sashrika) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java]: @harsha89  @surangak  i changed the openmrs formatter.

Window >> Preferences >> Java >> Code Style >> Formatter in the Control Statements tab:
Insert a new line before 'catch' in a 'try' statement
Insert a new line before 'finally' in a 'try' statement

Now the try catch new line is fixed. But why are having such wrong style in openmrs formatter?

There are several placed in fhir code base where the try catch spaces are not properly done. Do we need to change all lines? 

Editer>save options>edit all lines will work for this.

**Pull request merged**